### ISSUE TITLE
libbacktrace could not find executable to open

### ISSUE THREAD

**Issue by paradajz** [osx-cross/homebrew-avr#229]
Running on macOS Catalina. Not sure what else to provide here.

```
Building: board/gen/mega2560/Pins.cpp
cc1plus: internal compiler error: Illegal instruction: 4
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://github.com/osx-cross/homebrew-avr/issues> for instructions.
make[1]: *** [build/app/mega2560/release/board/gen/mega2560/Pins.cpp.o] Error 1
```

```
avr-gcc (Homebrew AVR GCC 9.3.0_2) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by ladislas** [comment]
What's the output of `brew doctor`

Can you try `brew remove avr-gcc avr-binutils && brew upgrade && brew install avr-gcc`?

**Comment by paradajz** [comment]
doctor output:
```

paradajzs-MacBook-Pro:src paradajz$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.profile

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/bin/code
```

Nothing changed after running what you suggested.

**Comment by ladislas** [comment]
Was it working before? Can you try on another project? 

Can you build from source?

Can you share your project so that I can check locally?

**Comment by paradajz** [comment]
Well, I'm running Catalina on MacBook Pro 2010 (installed through dosdude1 patch, if it matters - it was running clean install of High Sierra fist, then I upgraded it to Catalina). It worked few months ago before the upgrade - I'm not really using this laptop actively. The project is hosted here: https://github.com/paradajz/OpenDeck

How can i build the toolchain from source?

**Comment by ladislas** [comment]
> How can i build the toolchain from source?

`brew remove avr-gcc avr-binutils && brew install avr-binutils --head && brew install avr-gcc --head`

**Comment by paradajz** [comment]
I'm getting:

    Error: No head is defined for osx-cross/avr/avr-binutils
    Error: No head is defined for osx-cross/avr/avr-gcc@9

**Comment by ladislas** [comment]
ah right... I removed that...

Do you have another mac where you could test? A little more recent?

**Comment by paradajz** [comment]
Unfortunately I don't.

**Comment by ladislas** [comment]
I've just tested on my machine, it works great, I don't have the same issue.

Can you try another version? @8 or @10 to see if it's related?

**Comment by paradajz** [comment]
Both 8 and 10 act the same. I'm going to assume that is is something related to unofficial update to Catalina.

**Comment by ladislas** [comment]
Is your Xcode up to date?

**Comment by paradajz** [comment]
Yes.

**Comment by paradajz** [comment]
This could be of relevance:
https://github.com/conda/conda/issues/9678

I assume the binaries you provide are compiled on newer machines which probably have newer CPU features, which my Mac doesn't.

**Comment by ladislas** [comment]
0ed0db6 adds back the ability to build from source with `--head`.

Can you try again and let me know? Compilation can take between 30min and 3~4 hours depending on your machine. You can use the `-v` flag to see the output and make sure it's not stuck.

**Comment by paradajz** [comment]
I've tried with `brew install avr-gcc --build-from-source` flag an hour ago and it failed with same Illegal instruction 4 error after a while. I took it as a definite answer to "will Catalina work fine on this old MacBook" (it won't). I've reverted back to 10.13 High Sierra and running the install right now, will let you know. Since most of the things are now built from source (10.13 isn't officially supported by Homebrew anymore) it will take a while.
